pkg/controller/namespace/namespace_controller.go

```diff
@@ -209,25 +209,58 @@ func deleteAllContent(kubeClient client.Interface, experimentalMode bool, namesp
 	return estimate, nil
 }
 
-// syncNamespace makes namespace life-cycle decisions
-func syncNamespace(kubeClient client.Interface, experimentalMode bool, namespace api.Namespace) (err error) {
-	if namespace.DeletionTimestamp == nil {
-		return nil
+// updateNamespaceFunc is a function that makes an update to a namespace
+type updateNamespaceFunc func(kubeClient client.Interface, namespace *api.Namespace) (*api.Namespace, error)
+
+// retryOnConflictError retries the specified fn if there was a conflict error
+// TODO RetryOnConflict should be a generic concept in client code
+func retryOnConflictError(kubeClient client.Interface, namespace *api.Namespace, fn updateNamespaceFunc) (result *api.Namespace, err error) {
+	result = namespace
+	latestNamespace := namespace
+	for true {
+		result, err = fn(kubeClient, latestNamespace)
+		if err == nil {
+			break
+		}
+		if !errors.IsConflict(err) {
+			return nil, err
+		}
+		latestNamespace, err = kubeClient.Namespaces().Get(latestNamespace.Name)
+		if err != nil {
+			return nil, err
+		}
 	}
-	glog.V(4).Infof("Syncing namespace %s", namespace.Name)
+	return result, nil
```

Comment:
If you remove this the for loop is more idiomatic - for forever and all exits are returns